lib/Doctrine/DBAL/Driver/SQLite3/SQLite3Abstract.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\DBAL\Driver\SQLite3;
+
+/**
+ * Base functionality for SQLite3Connection and SQLite3Statement.
+ *
+ * @since 2.6
+ * @author Ben Morel <benjamin.morel@gmail.com>
+ * @author Bill Schaller <bill@zeroedin.com>
+ */
+abstract class SQLite3Abstract
+{
+    /**
+     * @var \SQLite3
+     */
+    protected $sqlite3;
+
+    /**
+     * @return void
+     *
+     * @throws SQLite3Exception
+     */
+    protected function throwExceptionOnError()
```

Comment:
It depends on what's decided. See my comments [here](https://github.com/doctrine/dbal/pull/2264#issuecomment-165269382) and [here](https://github.com/doctrine/dbal/pull/2264#issuecomment-165407106).